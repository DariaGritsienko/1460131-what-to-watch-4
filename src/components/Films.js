import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';

export default class Films extends PureComponent {
  constructor(props) {
    super(props);
  }

  onFilmsTitleClick(e, title) {
    const {onFilmsTitleClick} = this.props;
    onFilmsTitleClick(title);
  }

  render() {
    const {films} = this.props;
    return (
      <React.Fragment>
        {films.slice(1).map((film, index) =>
          <article
            className="small-movie-card catalog__movies-card"
            key={index}
          >
            <div className="small-movie-card__image">
              <img src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg" alt="Fantastic Beasts: The Crimes of Grindelwald" width="280" height="175" />
            </div>
            <h3 className="small-movie-card__title">
              <Link className="small-movie-card__link" onClick={(e) => this.onFilmsTitleClick(e, film.title)} to="/about">{film.title}</Link>
            </h3>
          </article>
        )}
      </React.Fragment>
    );
  }
}

Films.propTypes = {
  films: PropTypes.array,
  onFilmsTitleClick: PropTypes.func,
};
