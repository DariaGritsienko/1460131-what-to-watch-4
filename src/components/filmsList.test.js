import React from "react";
import renderer from "react-test-renderer";
import Films from './Films';

const titles = [
  `Some film1`, `Some film2`, `Some film3`
];

describe(`Foo`, () => {
  it(`<FilmsList /> should render cards with titles`, () => {
    const onFilmsTitleClick = jest.fn();
    const onMouseCard = jest.fn();

    const tree = renderer.create(
        <Films
          titles={titles}
          onFilmsTitleClick={onFilmsTitleClick}
          onMouseCard={onMouseCard}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

