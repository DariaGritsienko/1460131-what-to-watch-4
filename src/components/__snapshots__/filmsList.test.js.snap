// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo <FilmsList /> should render cards with titles 1`] = `null`;
